backend/interview_portal/src/main/java/by/interview/portal/controller/DisciplineController.java

```diff
@@ -16,39 +15,48 @@
 import org.springframework.web.bind.annotation.RestController;
 
 import by.interview.portal.domain.Discipline;
+import by.interview.portal.dto.DisciplineDTO;
+import by.interview.portal.dto.JwtUserDTO;
 import by.interview.portal.facade.DisciplineFacade;
 
 @CrossOrigin
 @RestController
 @RequestMapping(value = "/discipline")
 public class DisciplineController {
 
-	@Autowired
-	private DisciplineFacade disciplineFacade;
-
-	@ResponseStatus(value = HttpStatus.OK)
-	@GetMapping
-	public List<Discipline> findAll() {
-		return disciplineFacade.findByParentId(null);
-	}
-
-	@ResponseStatus(value = HttpStatus.OK)
-	@GetMapping(value = "/{id}")
-	public List<Discipline> findSubItems(@PathVariable Long id) {
-		return disciplineFacade.findByParentId(id);
-	}
-
-	@ResponseStatus(value = HttpStatus.OK)
-	@PostMapping
-	public void save(@RequestBody Discipline discipline) {
-		disciplineFacade.save(discipline);
-	}
-
-	@ResponseStatus(value = HttpStatus.OK)
-	@GetMapping(value = "/user")
-	public List<Discipline> findDisciplinesForUser() {
-		return disciplineFacade.findDisciplinesByUser(
-				((UserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal()).getUsername());
-	}
+    @Autowired
+    private DisciplineFacade disciplineFacade;
+
+    @ResponseStatus(value = HttpStatus.OK)
+    @GetMapping(value = "/{id}")
+    public DisciplineDTO findById(@PathVariable Long id) {
+        return disciplineFacade.findById(id);
+    }
+
+    @ResponseStatus(value = HttpStatus.OK)
+    @GetMapping
+    public List<Discipline> findAll() {
+        return disciplineFacade.findByParentId(null);
+    }
+
+    @ResponseStatus(value = HttpStatus.OK)
+    @GetMapping(value = "/parents/{id}")
+    public List<Discipline> findSubItems(@PathVariable Long id) {
+        return disciplineFacade.findByParentId(id);
+    }
+
+    @ResponseStatus(value = HttpStatus.OK)
+    @PostMapping
+    public void save(@RequestBody DisciplineDTO disciplineDTO) {
+        disciplineFacade.save(disciplineDTO);
+    }
+
+    @ResponseStatus(value = HttpStatus.OK)
+    @GetMapping(value = "/user")
+    public List<Discipline> findDisciplinesForUser() {
+        return disciplineFacade.findDisciplinesByUser(
+                ((JwtUserDTO) SecurityContextHolder.getContext().getAuthentication().getPrincipal())
+                        .getUsername());
+    }
 
```

Comment:
majority controllers return entity to the frontend, it needs to avoid

Reply:
In this case entity contains data that necessary to return to frontend. It doesn't make sense to create DTO with the same fields